src/frontends/tensorflow_common/src/op/reverse_sequence.cpp

```diff
@@ -14,14 +19,45 @@ namespace frontend {
 namespace tensorflow {
 namespace op {
 OutputVector translate_reverse_sequence_op(const NodeContext& node) {
-    default_op_checks(node, 2, {"ReverseSequence"});
+    default_op_checks(node, 2, {"ReverseSequence"}, true);
     auto input = node.get_input(0);
     auto seq_lengths = node.get_input(1);
 
     // retrieve attributes
     auto seq_dim = node.get_attribute<int64_t>("seq_dim");
     auto batch_dim = node.get_attribute<int64_t>("batch_dim", 0);
 
+    // handling negative values
+    if (seq_dim < 0) {
+        seq_dim -= 1;
+    }
+    if (batch_dim < 0) {
+        batch_dim -= 1;
+    }
+
+    auto complex_type_mark = as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr());
+    if (complex_type_mark) {
+        auto base_input = complex_type_mark->input_value(0);
+        // Split into real and imaginary parts
+        auto gather_index_real = make_shared<v0::Constant>(element::i32, Shape{}, 0);
+        auto gather_index_imag = make_shared<v0::Constant>(element::i32, Shape{}, 1);
+        auto input_real =
+            make_shared<v8::Gather>(base_input, gather_index_real, v0::Constant::create(element::i32, Shape{}, {-1}));
+        auto input_imag =
+            make_shared<v8::Gather>(base_input, gather_index_imag, v0::Constant::create(element::i32, Shape{}, {-1}));
+
+        // Reverse sequence for real and imaginary parts
+        auto reversed_real = make_shared<v0::ReverseSequence>(input_real, seq_lengths, batch_dim, seq_dim);
+        auto reversed_imag = make_shared<v0::ReverseSequence>(input_imag, seq_lengths, batch_dim, seq_dim);
+
+        // Recombine into a complex tensor
+        auto complex_concat = make_shared<v0::Concat>(OutputVector{reversed_real, reversed_imag}, -1);
```

Comment:
not needed to split into real and imag parts. Please pass `base_input` to `ReverseSequence`

Reply:
Okay will resolve it